pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java

```diff
@@ -265,6 +265,12 @@ protected static void validateUpsertConfig(TableConfig tableConfig, Schema schem
         tableConfig.getRoutingConfig() != null && RoutingConfig.REPLICA_GROUP_INSTANCE_SELECTOR_TYPE
             .equalsIgnoreCase(tableConfig.getRoutingConfig().getInstanceSelectorType()),
         "Upsert table must use replica-group (i.e. replicaGroup) based routing");
+    // no startree index
+    if (tableConfig.getIndexingConfig() != null && tableConfig.getIndexingConfig().getStarTreeIndexConfigs() != null) {
```

Comment:
This part is incorrect. You should wrap everything into the preconditions

Reply:
why? when both are null, this check is noop?

Reply:
If `tableConfig.getIndexingConfig().getStarTreeIndexConfigs()` is `null`, but `tableConfig.getIndexingConfig().isEnableDefaultStarTree()` is `true`, you will miss the check.
Also, `tableConfig.getIndexingConfig()` can never be `null` here because it has been checked

Reply:
I see what you meant. Makes sense.